pkg/controller/stack/stack_controller.go

```diff
@@ -260,28 +261,30 @@ func (r *ReconcileStack) Reconcile(ctx context.Context, request reconcile.Reques
 		return reconcile.Result{}, err
 	}
 
-	// these are used in status, and are set from some property of the source, whether it's the
-	// actual commit, or some analogue.
+	// This value is reported in .status, and is set from some property of the source -- whether
+	// it's the actual commit, or some analogue.
 	var currentCommit string
 
 	// Step 1. Set up the workdir, select the right stack and populate config if supplied.
 
-	// Check which kind of source we have. Since GitSource is not "omitempty", so its fields can be
-	// inline, it will be an empty value rather than nil when not supplied.
-	if stack.GitSource != nil {
-		// Ensure either branch or commit has been specified in the stack CR if stack is not marked for deletion
-		if stack.GitSource.ProjectRepo == "" || (stack.GitSource.Commit == "" && stack.GitSource.Branch == "") {
+	// Check which kind of source we have.
+	gitSource, fluxSource := stack.GitSource, stack.FluxSource
+	switch {
+	case gitSource != nil && fluxSource == nil:
```

Comment:
This logic is fine, but an alternative would be validating the mutually-exclusive condition ahead of the switch, and then keeping the simpler non-nil selection logic that you had previously. It probably doesn't matter for this PR, but it might be a little cleaner once the inline YAML case is added as well.